### PR TITLE
use external weather station safety logic via script or curl and support hysteresis on soft errors

### PR DESCRIPTION
**Pull request by d33psky**

### CODE REVIEW

**Comment by knro** [comment]
Great work Hans! Can we get some documentation/guide using the driver as well with screenshots?

**Comment by d33psky** [comment]
Thanks. Yes definitely. I'll make an entry in https://indilib.org/devices/weather-stations.html with screenshots. I'm also thinking about providing two example scripts. A bit like libindi/drivers/dome/dome_script.txt but unsure where to put those. Then there's the 'weather safety service' itself which this driver talks to. I want to help people build one of their own, and what the driver expects the service provides.

File: libindi/drivers/weather/weather_safety_proxy.cpp [hunk 1]
```diff
@@ -0,0 +1,474 @@
...
+    setVersion(1, 0);
+
+    setWeatherConnection(CONNECTION_NONE);
+}
+
+WeatherSafetyProxy::~WeatherSafetyProxy() {}
+
+const char *WeatherSafetyProxy::getDefaultName()
+{
+    return (const char *)"Weather_Safety_Proxy";
```
Comment (knro) [inline libindi/drivers/weather/weather_safety_proxy.cpp]: Do not use _ in the name. Also no need for (const char *)

**Comment by knro** [comment]
Thanks, can you attach the screenshots? I couldn't find them in the submission.

**Comment by d33psky** [comment]
i'm still working on the weather-stations/weather-safety-proxy.html

**Comment by d33psky** [comment]
> i'm still working on the weather-stations/weather-safety-proxy.html
This is now done. https://www.indilib.org/devices/weather-stations/weather-safety-proxy.html is ready.

**Comment by knro** [comment]
Is this PR ready or you need to include more changes?

**Comment by d33psky** [comment]
it's ready to be merged into master

**Pull request merged**